lib/IlmCtl/CtlRcPtr.cpp

```diff
@@ -96,7 +96,7 @@ throwRcPtrExc (const RcObject *lhs, const RcObject *rhs)
 Mutex &
 rcPtrMutex (RcObject *ptr)
 {
-    return mutexes[reinterpret_cast<unsigned long>(ptr) % NUM_MUTEXES];
+    return mutexes[reinterpret_cast<unsigned long long int>(ptr) % NUM_MUTEXES];
```

Comment:
@AntonQubecinema I've checked this on 32-bit `Linux ubuntu 4.4.0-53-generic #74-Ubuntu SMP Fri Dec 2 15:58:04 UTC 2016 i686 i686 i686 GNU/Linux` and included tests seem to pass